internal/postgres/config.go

```diff
@@ -201,6 +205,46 @@ func startupCommand(
 	version := fmt.Sprint(cluster.Spec.PostgresVersion)
 	walDir := WALDirectory(cluster, instance)
 
+	// If the user requests tablespaces, we want to make sure the directories exist with the
+	// correct owner and permissions.
+	tablespaceCmd := ""
+	if util.DefaultMutableFeatureGate.Enabled(util.TablespaceVolumes) && instance.TablespaceVolumes != nil {
+		// This command checks if a dir exists and if not, creates it;
+		// if the dir does exist, then we `recreate` it to make sure the owner is correct;
+		// but if the dir exists with the wrong owner and is not writeable, we error.
+		// This is the same behavior we use for the main PGDATA directory.
+		// Note: Postgres requires the tablespace directory to be "an existing, empty directory
+		// that is owned by the PostgreSQL operating system user."
+		// - https://www.postgresql.org/docs/current/manage-ag-tablespaces.html
+		// However, unlike the PGDATA directory, Postgres will _not_ error out
+		// if the permissions are wrong on the tablespace directory.
+		// Instead, when a tablespace is created in Postgres, Postgres will
+		// chmod the tablespace directory to match permissions on the PGDATA directory,
+		// i.e., 700 or 750
+		// Postgres setting the directory permissions:
+		// - https://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/backend/commands/tablespace.c;h=5411638696b4d1436a56f62e9cde7ec61797377b;hb=REL_14_0#l604
+		// Postgres setting directory permissions:
+		// - https://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/common/file_perm.c;h=ab85765a0bfb5108265819a4c7e9ff87dc6add10;hb=REL_14_0#l34
```

Comment:
⛏️ GitWeb URLs are long, but parts are redundant and can be omitted for brevity.

🤔 The two links have the same English above them but do two different things. Perhaps the 2nd explanation is unintentional?

```suggestion
		// Postgres setting the tablespace directory permissions:
		// - https://git.postgresql.org/gitweb/?p=postgresql.git;f=src/backend/commands/tablespace.c;hb=REL_14_0#l600
		// Postgres choosing directory permissions:
		// - https://git.postgresql.org/gitweb/?p=postgresql.git;f=src/common/file_perm.c;hb=REL_14_0#l27
```

Reply:
Heh, the first "setting" means "this is where PG sets the dir permissions"; the second "setting" is "this is where PG sets the internal variable it uses" -- I could've found another way to say that...